lib/rules/no-implicit-coercion.js

```diff
@@ -246,7 +248,13 @@ module.exports = {
                 // ~foo.indexOf(bar)
                 operatorAllowed = options.allow.indexOf("~") >= 0;
                 if (!operatorAllowed && options.boolean && isBinaryNegatingOfIndexOf(node)) {
-                    const recommendation = `${sourceCode.getText(node.argument)} !== -1`;
+
+                    /*
+                     * `foo?.indexOf(bar) !== -1` will be true (== found) if the `foo` is nullish. So use `>= 0` in that case.
+                     * Or maybe we should recommend `foo?.includes(bar)` simply.
+                     */
+                    const comparison = node.argument.type === "ChainExpression" ? ">= 0" : "!== -1";
+                    const recommendation = `${sourceCode.getText(node.argument)} ${comparison}`;
```

Comment:
I think `>= 0` is fine for this rule because recommending `includes` might be beyond its responsibility, and it can't know if the user wants to use ES2016+ array methods.